src/custom/conversions.rs

```diff
@@ -0,0 +1,246 @@
+use super::{CodegenExtension, CodegenExtsMap};
+
+use anyhow::{anyhow, Result};
+
+use hugr::{
+    extension::{
+        prelude::{sum_with_error, ConstError},
+        simple_op::MakeExtensionOp,
+        ExtensionId,
+    },
+    ops::{constant::Value, custom::ExtensionOp},
+    std_extensions::arithmetic::{
+        conversions::{self, ConvertOpDef},
+        int_types::INT_TYPES,
+    },
+    types::CustomType,
+    HugrView,
+};
+
+use inkwell::types::BasicTypeEnum;
+use inkwell::values::{AnyValue, BasicValue};
+
+use crate::{
+    emit::{
+        func::EmitFuncContext,
+        ops::{emit_custom_unary_op, emit_value},
+        EmitOp, EmitOpArgs,
+    },
+    sum::LLVMSumType,
+    types::TypingSession,
+};
+
+struct ConversionsEmitter<'c, 'd, H>(&'d mut EmitFuncContext<'c, H>);
+
+fn log2(m: u32) -> Option<u8> {
+    if m == 1 {
+        Some(0)
+    } else if m % 2 == 0 {
+        let n = log2(m / 2)?;
+        Some(n + 1)
+    } else {
+        None
+    }
+}
```

Comment:
I think you won't actually need this function (see my comment above), but just in case you can do this more efficiently via `32 - m.leading_zeros() - 1`

Reply:
🤯